modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/ManagerDeploymentDirectoryWatcher.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.deployment.internal;
+
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+import static org.mule.runtime.api.util.MuleSystemProperties.DEPLOYMENT_APPLICATION_PROPERTY;
+import static org.mule.runtime.api.util.MuleSystemProperties.SINGLE_DEPLOYMENT_PROPERTY;
+import static org.mule.runtime.module.deployment.internal.MuleDeploymentService.PARALLEL_DEPLOYMENT_PROPERTY;
+import static org.mule.runtime.module.deployment.internal.MuleDeploymentService.useParallelDeployment;
+
+import static java.lang.String.format;
+import static java.lang.System.getProperties;
+import static java.lang.System.getProperty;
+
+import org.mule.runtime.api.scheduler.SchedulerService;
+import org.mule.runtime.deployment.model.api.application.Application;
+import org.mule.runtime.deployment.model.api.domain.Domain;
+import org.mule.runtime.module.artifact.api.descriptor.ApplicationDescriptor;
+import org.mule.runtime.module.artifact.api.descriptor.DomainDescriptor;
+import org.mule.runtime.module.deployment.internal.util.ObservableList;
+
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Supplier;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Manages the deployment directory needed.
+ *
+ * @since 4.7.0
+ */
+public class ManagerDeploymentDirectoryWatcher {
```

Comment:
This should be named as a provider or something like that. This is not a watcher but obtains the watcher.